example_usage.py

```diff
@@ -0,0 +1,132 @@
+from datasets import load_dataset
+import numpy as np
+from PIL import Image
+import torch as th
+from torch.utils.data import DataLoader
+
+from simple_transformers.transformer import ModalityEncoder, ModalityEncoderDecoder
+
+
+def classification_train_loop(model, dataset, input_key, use_pil_images=False):
+    optimizer = th.optim.Adam(model.parameters(), lr=1e-4)
+    dataloader = DataLoader(dataset, batch_size=64, shuffle=True)
+    loss_fn = th.nn.CrossEntropyLoss()
+    for epoch in range(10):
+        losses, accuracies = [], []
+        for batch in dataloader:
+            optimizer.zero_grad()
+
+            if use_pil_images:
+                model_input = [Image.fromarray(i).convert('RGB') for i in batch[input_key].numpy()]
+            elif input_key == 'image' and len(batch[input_key].shape) == 3:
+                model_input = batch[input_key].unsqueeze(1).numpy()
+            else:
+                model_input = batch[input_key]
+            logits = model(model_input)
+            _, preds = th.max(logits, dim=-1)
+            loss = loss_fn(logits, batch['label'])
+            loss.backward()
+            optimizer.step()
+            losses += [loss.item()]
+            accuracies += [th.mean((preds == batch['label']).float()).item()]
+        print(f'LOSS: {np.mean(losses)}, ACCURACY: {np.mean(accuracies)}')
+
+
+def text_gen_train_loop(model, dataset):
+    optimizer = th.optim.Adam(model.parameters(), lr=1e-4)
+    dataloader = DataLoader(dataset, batch_size=64)
+    loss_fn = th.nn.CrossEntropyLoss()
+    for epoch in range(10):
+        losses, accuracies = [], []
+        in_string, out_string, gen_string = None, None, None
+        for batch in dataloader:
+            optimizer.zero_grad()
+            logits, gen_strings = model(batch['translation']['en'], batch['translation']['fr'])
+            in_string, out_string, gen_string = batch['translation']['en'][0], batch['translation']['fr'][0], \
+            gen_strings[0]
+            _, preds = th.max(logits, dim=-1)
+            labels = model.output_preprocessor.tokenizer(batch['translation']['fr'], padding=True, return_tensors='pt')[
+                'input_ids']
+            batch_size, seq_len = labels.shape
+            # TODO: Loss should be masked too
```

Comment:
nit: Remove TODO or still valid?

Reply:
still valid. I don't have time to fix example usage rn, but will put it as a todo  in the readme